### DISCUSSION TITLE
Galaxy Lamp master switch

### DISCUSSION THREAD

**Discussion by GTunney** [make-all/tuya-local#2428] [Q&A]
Does the master switch being part of the configuration and not controls within home assistant effect its functionality at all?

I expose all my devices to Alexa but it will not discover the master switch. It will discover the background and laser entities. I have numerous other switch devices which are also discovered. Just not the master switch. 

It is also not possible to turn on or off the switch when doing by area. Using dev tools I can’t run the action switch.turnon and switch.turnoff based on area.

**Comment by make-all** [comment]
Exposure to voice assistants by default is one of the things that category tags affect. I think it can be manually exposed, but this is beyond the scope of this integration so you'll have to check the HA docs for how to do that.

**Reply by GTunney** [reply]
Thanks, any idea on why it won’t work turning on or off by area?

**Reply by make-all** [reply]
No idea. Is the entity in the Area? Maybe entities marked as config are not added by default when the device is added to the area, same as what happens on the UI and voice assistants.

**Reply by GTunney** [reply]
Yup it's in the area. 

Is there a reason why the master switch was chosen to be added as a config and not a control by TuyaLocal?

Could this behaviour be changed so it can be added as a control?

**Comment by GTunney** [comment]
I have managed to resolve this by removing the category: config from the master switch within devices/moes_star_projector.yaml

This now allows me to control the star lamp via area and is also now discoverable within Alexa.

I'm not sure why the decision was made to have the master switch as a config category. It makes more sense to me to have it as a control given that's what it is doing, controlling the the overall projector.

**Comment by make-all** [comment]
Probably the decision had something to do with this comment on the original request:
https://github.com/make-all/tuya-local/issues/821#issuecomment-1877816853

> Though having a master switch could sort of be redundant as dashboard entries have a master (screenshot 1). 

I will change the config, since I'm also not sure it makes sense to not have it as a primary control.

**Reply by GTunney** [reply]
> Probably the decision had something to do with this comment on the original request: [#821 (comment)](https://github.com/make-all/tuya-local/issues/821#issuecomment-1877816853)
> 
> > Though having a master switch could sort of be redundant as dashboard entries have a master (screenshot 1).
> 
> I will change the config, since I'm also not sure it makes sense to not have it as a primary control.

Did this change get missed in the latest update I'm guessing?

Just updated and it's reverted master switch back to a config not control item.
